src/transaction_builder.cpp

```diff
@@ -262,7 +262,9 @@ TransactionBuilder::TransactionBuilder(
     cs_coinsView(cs_coinsView),
     orchardAnchor(orchardAnchor)
 {
-    mtx = CreateNewContextualCMutableTransaction(consensusParams, nHeight);
+    mtx = CreateNewContextualCMutableTransaction(
+            consensusParams, nHeight,
+            !orchardAnchor.has_value() && nPreferredTxVersion < ZIP225_MIN_TX_VERSION);
```

Comment:
Document the precondition on `orchardAnchor` that is required for this logic to work.